### DISCUSSION TITLE
Disable to automatically reestablish PDU Session if release from CLI

### DISCUSSION THREAD

**Discussion by bojanHub** [aligungr/UERANSIM#491] [Q&A]
As I see in src/ue/nas/sm/resource.cpp - NasSm::establishRequiredSessions -  PDU session is automatically generated if it is configured in config->defaultSessions and currently is PDU session not established.
I like it for first PDU session creation after UE registration but if I stop PDU session from CLI I would not like to established it again automatically. 

Is there any option/configuration that in the second case (after CLI PDU release) PDU session will not be established again?

Currently for test I change source code in the following way to delete defaultSessions after creation of sessions to prevent PDU session to be established again but I would prefer some official solution:
```
void NasSm::establishRequiredSessions()
{
...
    **int isEstablished = 0;**
    for (auto &config : m_base->config->defaultSessions)
    {
        if (!anySessionMatches(config)) {
            sendEstablishmentRequest(config);
            **isEstablished = 1;**
        }
    }
    **if (isEstablished) {
        m_base->config->defaultSessions.clear();
    }**
}
```

**Comment by shashidhar-patil** [comment]
Hey @bojanHub, did you find any work around ?

**Comment by bojanHub** [comment]
Hi @shashidhar-patil, 

my workaround was what I have already posted - additional code is inside marks **. I just deleted defaultSessions after PDU session is established. On this way new session is not automatically reestablished.

But I didn't check if there were any changes in ueransim code in the last year.

**Reply by shashidhar-patil** [reply]
Great, Thanks!
